x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -621,6 +621,23 @@ function isValidTimeInterval(value: string | undefined): boolean {
   return parseTimeIntervalForJob(value) !== null;
 }
 
+// The earliest start time for the datafeed should be the max(latest_record_timestamp, latest_bucket.timestamp + bucket_span).
+export function getEarliestDatafeedStartTime(
```

Comment:
Would be good to add a unit test for this new function too.

Reply:
Added tests in https://github.com/elastic/kibana/pull/71976/commits/dcce6b69503b4cb2e2252e705097a3657dc776dc